src/main/java/berlin/yuna/nano/services/http/model/HttpRequest.java

```diff
@@ -0,0 +1,305 @@
+                            final String value = java.net.URLDecoder.decode(keyValue[1], Charset.defaultCharset());
+                            queryParams.put(key, value);
+                        } else if (keyValue.length == 1) {
+                            final String key = java.net.URLDecoder.decode(keyValue[0], Charset.defaultCharset());
+                            queryParams.put(key, "");
+                        }
+                    }
+                }
+            } catch (final Exception ignored) {
+            }
+        }
+        return queryParams;
+    }
+
+    public boolean containsQueryParam(final String key) {
+        return queryParams.containsKey(key);
+    }
+
+    public String getQueryParam(final String key) {
+        final Object value = queryParams.get(key);
+        return value != null ? value.toString() : null;
+    }
+
+    public boolean exactMatch(final String path) {
+        return this.path.startsWith(path);
+    }
+
+    public boolean match(final String pathToMatch) {
+
+        final String[] partsToMatch = pathToMatch.split("/");
+        final String[] parts = path.split("/");
+
+        if (exactMatch(pathToMatch) && pathToMatch.contains("{")) return true;
+
+        if (partsToMatch.length != parts.length) return false;
+
+        if (pathParams == null)
+            pathParams = new TypeMap();
+        else
+            pathParams.clear();
+        for (int i = 0; i < partsToMatch.length; i++) {
+            if (!partsToMatch[i].equals(parts[i])) {
+                if (partsToMatch[i].startsWith("{")) {
+                    final String key = partsToMatch[i].substring(1, partsToMatch[i].length() - 1);
+                    pathParams.put(key, parts[i]);
+                } else {
+                    return false;
+                }
+            }
+        }
+
+        return true;
+    }
+
+
+    public TypeMap getPathParams() {
+        return pathParams;
+    }
+
+    public String getPathParam(final String key) {
+        return pathParams.get(key).toString();
+    }
+
+    public String getHeader(final String name) {
+        String value = headers.get(name).toString();
+        if (value.startsWith("[")) {
+            value = value.substring(1, value.length() - 1);
+        }
+        return value;
+    }
+
+    public TypeMap getHeaders() {
+        return headers;
+    }
+
+    public boolean containsHeader(final String name) {
+        return headers.containsKey(name.toLowerCase());
+    }
+
+    public String host() {
+        return exchange.getLocalAddress().getHostName();
+    }
+
+    public int port() {
+        return exchange.getLocalAddress().getPort();
+    }
+
+    public String protocol() {
+        return exchange.getProtocol();
+    }
+
+    public String userAgent() {
+        return getHeader(HttpHeaders.USER_AGENT);
+    }
+
+    public String getAuthorisationToken() {
```

Comment:
1) please no `get` or `set`
2) make it shorter like `authToken()` - users should not read books when chaining these methods
3) add a method like `basicAuth()` which returns a decoded user/password _(maybe returned as String[]{user, password})_